### DISCUSSION TITLE
How to configure IP addresses for IP Pools?

### DISCUSSION THREAD

**Discussion by bakgaard** [postalserver/postal#1447] [Help with using Postal]
According to [the documentation](https://github.com/postalhq/postal/wiki/Configuring-IP-Pools#configuring-ips-on-server), I must setup my server to handle extra IPs I want to use in my IP Pools.

What does this mean?

I take I have to do something like adding it to ``/etc/network/interfaces``, but I have absolutely no idea what I need to do about it.
I am running Postal through ``docker-compose`` which might only make this a bit more tricky.

Quick note for the following - I have external IPs, and I have an internal IP.
My external IP1 and IP2 will for this example be `123.123.123.1` and `123.123.123.2`.
My internal IP will be `10.10.10.1`.

When you hit ``postal.mysite.com`` you actually hit `123.123.123.1`, and it will forward you to `10.10.10.1`.
From here Nginx will send you into my container.
I have setup `123.123.123.2` to also forward to `10.10.10.1`

I would like to add `123.123.123.2` to my IP Pool - what do I do?

If I run the command ``ifconfig`` on my host, I get something like this:

```
br-50e152d57bed: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.22.0.1  netmask 255.255.0.0  broadcast 172.22.255.255

docker0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.0.0  broadcast 172.17.255.255

ens160: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.10.10.1  netmask 255.255.255.0  broadcast 10.10.10.255

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
...
```

**Comment by willpower232** [comment]
The IP addresses will need to come from your server provider. They will usually have instructions for how to configure them as they will have specific network settings. You should ask them first.

**Comment by creative-andrew** [comment]
I am using Contabo as my VPS ( I am not affiliated with them and I don't endorse them), but they have "great" documentation (or least beginner-friendly ) regarding setting up extra IPs.

https://contabo.com/blog/configuring-additional-ip-addresses/

**Comment by TuguldurJ** [comment]
Add interface like this: 

```
auto eth0:0
iface eth0:0 inet static
        address IP_address
        netmask 255.255.255.255
        post-up /sbin/ifconfig eth0:0 IP_address netmask 255.255.255.255 broadcast IP_address
        pre-down /sbin/ifconfig eth0:0 down
```

**Comment by ivansalloum** [comment]
For anyone still running into this, yes – you need to configure your host server to recognize the additional external IPs you want to use in Postal’s IP Pools (not just forward them internally). This usually means assigning them directly to the network interface.

I recently wrote a full step-by-step guide on setting up IP Pools in Postal, including how to configure extra public IPs properly (with Hetzner Floating IPs, but the principles apply broadly): [How to Use IP Pools for Better Deliverability in Postal](https://ivansalloum.com/how-to-use-ip-pools-in-postal/)

Might save you some trial and error!
